plenum/server/consensus/view_change_service.py

```diff
@@ -80,29 +80,29 @@ def process_need_view_change(self, msg: NeedViewChange):
         # 3. Update shared data
         self._data.view_no = view_no
         self._data.waiting_for_new_view = True
+        self._data.primary_name = None
         if not self._data.is_master:
-            self._data._master_reordered_after_vc = False
-        self._data.primaries = self._primaries_selector.select_primaries(view_no=self._data.view_no)
-        for i, primary_name in enumerate(self._data.primaries):
-            logger.display("{} selected primary {} for instance {} (view {})"
-                           .format(PRIMARY_SELECTION_PREFIX,
-                                   primary_name, i, self._data.view_no),
-                           extra={"cli": "ANNOUNCE",
-                                  "tags": ["node-election"]})
-
-        old_primary = self._data.primary_name
-        self._data.primary_name = generateName(self._data.primaries[self._data.inst_id], self._data.inst_id)
+            self._data.master_reordered_after_vc = False
 
         if not self._data.is_master:
             return
 
+        # Backup primaries will be re-selected after re-ordering is done
+        old_primary = self._data.primary_name
```

Comment:
:warning: It looks like `old_primary` here will always be None, was it intended?

Also I think it is worth mentioning that reselection of backup primaries will be needed just because F could change after a view change, otherwise it sounds like we're selecting some primaries, do a view change and then select completely different set primaries, which is actually not the case.

Reply:
Thanks for catching this, fixed.
It looks like the integration test that checked this logic  at the time of implementation (`test_promotion_before_view_change`) doesn't reproduce the situation when the same Primary is selected (since now we append added/promoted nodes to the end of the node reg list).
We need to think if the case is possible in general (that the same primary is selected). and create a new test. I think can be done in next PRs